### PR TITLE
minor code refinement: `if_exists` name change, wildcard field for logical plan, etc.

### PR DESCRIPTION
**Pull request by xudong963**
# Which issue does this PR close?

Code found during the code review process that could be refined.

### CODE REVIEW

File: datafusion/src/logical_plan/plan.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ pub struct DropTable {
     /// The table name
     pub name: String,
     /// If the table exists
-    pub if_exist: bool,
+    pub if_exists: bool,
```
Comment (xudong963) [inline datafusion/src/logical_plan/plan.rs:194]: Consistent with `Statement::Drop`

File: datafusion/src/sql/planner.rs [hunk 2]
```diff
@@ -194,7 +182,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             {
                 Ok(LogicalPlan::DropTable(DropTable {
                     name: names.get(0).unwrap().to_string(),
-                    if_exist: *if_exists,
+                    if_exists: *if_exists,
```
Comment (xudong963) [inline datafusion/src/sql/planner.rs:185]: After https://github.com/apache/arrow-datafusion/pull/1945/files merged,  here can directly `if_exists`

**Review by houqp — APPROVED** [review]

**Review by liukun4515 — APPROVED** [review]
LGTM

**Review by Jimexist — APPROVED** [review]
thank you

**Pull request merged**